### PR TITLE
Added pseudo api for file upload and data class

### PR DESCRIPTION
**Pull request by NLuffy**
Data class design is pending so only implemented it as a dummy. Also the api will connect with aws class which needs to be discussed.

### CODE REVIEW

File: src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package com.example.simplefileshare.simplefileshare.controller;
+import org.springframework.web.bind.annotation.*;
+import java.io.File;
+
+@RestController
+@RequestMapping("/api/upload")
```
Comment (shashank-vats) [inline src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java]: Can we rename it to api/files/?

File: src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+public class FileController {
+
+    @GetMapping("/api/get")
```
Comment (shashank-vats) [inline src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java]: No need to give endpoint here if we are using Rest mappings and giving an endpoint on the class itself

File: src/main/java/com/example/simplefileshare/simplefileshare/data/Filedata.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.example.simplefileshare.simplefileshare.data;
+
+
+public class Filedata {
```
Comment (shashank-vats) [inline src/main/java/com/example/simplefileshare/simplefileshare/data/Filedata.java]: We can use Lombok to automatically generate getters and setters

File: src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java (hunk 1 above)
Comment (NLuffy) [reply src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java]: Sure will modify it.

File: src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java (hunk 2 above)
Comment (NLuffy) [reply src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java]: Yes added the comment in it not needed already :p

File: src/main/java/com/example/simplefileshare/simplefileshare/data/Filedata.java (hunk 3 above)
Comment (NLuffy) [reply src/main/java/com/example/simplefileshare/simplefileshare/data/Filedata.java]: Okay. Will check it.

File: src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+
+@RestController
+@RequestMapping("/api/files")
+public class FileController {
+
+    private UploadService uploadService = new UploadService();
+
+    @PostMapping
+    public String upload(@RequestBody MultipartFile file) throws Exception{
```
Comment (shashank-vats) [inline src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java]: Handle exception as discussed

File: src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.web.multipart.MultipartFile;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+@RestController
+@RequestMapping("/api/files")
+public class FileController {
+
+    private UploadService uploadService = new UploadService();
```
Comment (shashank-vats) [inline src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java]: Use dependency injection

File: src/main/java/com/example/simplefileshare/simplefileshare/services/UploadService.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+                throw new FileNotFoundException("Invalid File");  
+        } catch(Exception e) {
+            Logger.getGlobal().log(Level.SEVERE, e.getMessage());
+        }
+        return result;
+    }
+
+    public File convertMultipartFileToFile(MultipartFile multipartFile) {
+        try{
+            File file = new File("/c/Users/nishekha/Desktop/Java/SimpleFileShare/src/main/java/com/example/simplefileshare/simplefileshare/aws/utils/file.tmp");
```
Comment (shashank-vats) [inline src/main/java/com/example/simplefileshare/simplefileshare/services/UploadService.java]: Use relative paths

File: src/main/java/com/example/simplefileshare/simplefileshare/services/utils/MultipartFileToJavaFileUtil.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
+package com.example.simplefileshare.simplefileshare.services.utils;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import java.io.FileNotFoundException;
+import java.io.File;
+
+@Service
+public class MultipartFileToJavaFileUtil {
+    
+    public File convert(MultipartFile multipartFile) throws FileNotFoundException {
```
Comment (shashank-vats) [inline src/main/java/com/example/simplefileshare/simplefileshare/services/utils/MultipartFileToJavaFileUtil.java]: Can we change this to a static method and remove service annotation?

File: src/main/java/com/example/simplefileshare/simplefileshare/services/utils/MultipartFileToJavaFileUtil.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+        try{
+            File file = new File(multipartFile.getName());
+            multipartFile.transferTo(file);
+            System.out.println("File Created Successfully");
+            return file;
+        }
+        catch(Exception e){
```
Comment (shashank-vats) [inline src/main/java/com/example/simplefileshare/simplefileshare/services/utils/MultipartFileToJavaFileUtil.java:20]: Can we throw the error exactly as it is and reserve FileNotFoundException to only those cases where file is not found in the request?

File: src/main/java/com/example/simplefileshare/simplefileshare/services/UploadInterface.java [hunk 9]
```diff
@@ -0,0 +1,10 @@
+package com.example.simplefileshare.simplefileshare.services;
+import java.io.File;
+import org.springframework.web.multipart.MultipartFile;
+import com.amazonaws.SdkClientException;
+import java.io.FileNotFoundException;
+
+public interface UploadInterface {
+    public String uploadFile (File file) throws SdkClientException;
+    public String uploadMultipartFile (MultipartFile multipartFile) throws SdkClientException,FileNotFoundException;
```
Comment (shashank-vats) [inline src/main/java/com/example/simplefileshare/simplefileshare/services/UploadInterface.java]: Please remove the exceptions from the interface as other implementations may not throw the same exceptions.

File: src/main/java/com/example/simplefileshare/simplefileshare/services/AWSService.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    @Override
+    public String uploadMultipartFile (MultipartFile multipartFile) throws FileNotFoundException, SdkClientException {
+            File file = multipartFileToJavaFileConverter.convert(multipartFile);
+            return this.uploadFile(file); 
+    }
+
+    @Override
+    public String uploadFile (File file) throws SdkClientException {
+        AWSObjectUtils objectUtils = new AWSObjectUtils(awsS3Client);
+        objectUtils.uploadFile("simplefilehost", "AWSBucketUtils", file, "plain/text");
```
Comment (shashank-vats) [inline src/main/java/com/example/simplefileshare/simplefileshare/services/AWSService.java]: How will we decide key name, needs to be discussed.

File: src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    @PostMapping
+    public String upload(@RequestBody MultipartFile file) throws Exception{
+        try{
+            if(!file) throw new FileNotFoundException("Invalid File");
+            String url = uploadService.uploadMultipartFile(file);
+            return new ResponseEntity<>(new BaseResponse<>(null, url), HttpStatus.CREATED);
+        } catch (SdkClientException e){
+            throw ErrorUtils.createApiError(HttpStatus.INTERNAL_SERVER_ERROR, HttpStatus.INTERNAL_SERVER_ERROR.getReasonPhrase(), "Some error occurred");
+        } catch (FileNotFoundException e){
+            throw ErrorUtils.createApiError(HttpStatus.INTERNAL_SERVER_ERROR, HttpStatus.INTERNAL_SERVER_ERROR.getReasonPhrase(), "Some file error occurred");
```
Comment (shashank-vats) [inline src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java:32]: We should throw a Bad request error in this case.

File: src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java [hunk 12]
```diff
@@ -0,0 +1,34 @@
+package com.example.simplefileshare.simplefileshare.controller;
```
Comment (shashank-vats) [inline src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java:1]: Can we keep only one package either `controller` or `controllers` and move both the example controller and FileController there?

File: src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java (hunk 5 above)
Comment (NLuffy) [reply src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java]: Updated in the next commit.

File: src/main/java/com/example/simplefileshare/simplefileshare/services/UploadService.java (hunk 6 above)
Comment (NLuffy) [reply src/main/java/com/example/simplefileshare/simplefileshare/services/UploadService.java]: Added tmp folder for this and handled it

File: src/main/java/com/example/simplefileshare/simplefileshare/services/utils/MultipartFileToJavaFileUtil.java (hunk 7 above)
Comment (NLuffy) [reply src/main/java/com/example/simplefileshare/simplefileshare/services/utils/MultipartFileToJavaFileUtil.java]: Yes its another way of doing this...

File: src/main/java/com/example/simplefileshare/simplefileshare/services/utils/MultipartFileToJavaFileUtil.java (hunk 8 above)
Comment (NLuffy) [reply src/main/java/com/example/simplefileshare/simplefileshare/services/utils/MultipartFileToJavaFileUtil.java:20]: okay will add it.

File: src/main/java/com/example/simplefileshare/simplefileshare/services/UploadInterface.java (hunk 9 above)
Comment (NLuffy) [reply src/main/java/com/example/simplefileshare/simplefileshare/services/UploadInterface.java]: okay.

File: src/main/java/com/example/simplefileshare/simplefileshare/services/AWSService.java (hunk 10 above)
Comment (NLuffy) [reply src/main/java/com/example/simplefileshare/simplefileshare/services/AWSService.java]: Sure.

File: src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java (hunk 12 above)
Comment (NLuffy) [reply src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java:1]: okay.

File: src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+@RestController
+@RequestMapping("/api/files")
+public class FileController {
+
+    @Autowired
+    private AWSService uploadService;
+
+    @PostMapping
+    public ResponseEntity<BaseResponse<String>> upload(@RequestParam("file") MultipartFile file) throws Exception{
+        System.out.println(file);
```
Comment (shashank-vats) [inline src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java]: Should we check here if file is null?

Comment (NLuffy) [reply src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java]: Added the check

**Review by shashank-vats — APPROVED** [review]

File: src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java (hunk 11 above)
Comment (NLuffy) [reply src/main/java/com/example/simplefileshare/simplefileshare/controller/FileController.java:32]: Added the changes.

**Pull request merged**